onnxruntime/test/contrib_ops/dynamic_quantize_matmul_test.cc

```diff
@@ -0,0 +1,204 @@
+    Y_data_.resize(Y_size);
+
+    AddInput<float>("A", A_dims_, A_data_);
+    AddInput<T>("B", B_dims_, B_data_);
+    AddInput<float>("b_scale", {1}, B_scale_);
+    AddInput<T>("b_zero_point", {1}, B_zero_point_);
+
+    AddOutput<float>("Y", Y_dims_, Y_data_);
+  }
+  void Run() {
+#ifndef NDEBUG
+    run_called_ = true;
+#endif
+    std::vector<MLValue> cpu_fetches;
+    std::vector<MLValue> subgraph_fetches;
+    Compute(cpu_fetches);
+    ComputeOriginalSubgraph(subgraph_fetches);
+
+    // Compare CPU with original subgraph on the output(0) - Y
+    ASSERT_TRUE(cpu_fetches.size() >= subgraph_fetches.size());
+    for (size_t i = 0; i < subgraph_fetches.size(); i++) {
+      if (cpu_fetches[i].IsTensor() && subgraph_fetches[i].IsTensor()) {
+        VLOGS_DEFAULT(1) << "Checking tensor " << i;
+        CheckTensor(subgraph_fetches[i].Get<Tensor>(), cpu_fetches[i].Get<Tensor>(), 1e-3, 1e-3);
+      }
+    }
+  }
+
+ private:
+  void Compute(std::vector<MLValue>& cpu_fetches) {
+    auto p_model = BuildGraph();
+    auto& graph = p_model->MainGraph();
+
+    Status status = graph.Resolve();
+    ASSERT_TRUE(status.IsOK()) << status;
+
+    // Hookup the inputs and outputs
+    std::unordered_map<std::string, MLValue> feeds;
+    std::vector<std::string> output_names;
+    FillFeedsAndOutputNames(feeds, output_names);
+
+    SessionOptions so;
+    so.session_logid = op_;
+    so.session_log_verbosity_level = 1;
+
+    RunOptions run_options;
+    run_options.run_tag = op_;
+    run_options.run_log_verbosity_level = 1;
+
+    // run with LayerNormalization
+    InferenceSession layernorm_session_object{so, GetEnvironment()};
+    std::string s1;
+    p_model->ToProto().SerializeToString(&s1);
+    std::istringstream str(s1);
+    ASSERT_TRUE((status = layernorm_session_object.Load(str)).IsOK()) << status;
+    ASSERT_TRUE((status = layernorm_session_object.Initialize()).IsOK()) << status;
+    ASSERT_TRUE((status = layernorm_session_object.Run(run_options, feeds, output_names, &cpu_fetches)).IsOK());
+  }
+
+  void ComputeOriginalSubgraph(std::vector<MLValue>& subgraph_fetches) {
+    NameMLValMap feeds;
+    OrtValue ml_value;
+    std::vector<std::string> output_names{"Y"};
+
+    CreateMLValue<float>(TestCPUExecutionProvider()->GetAllocator(0, OrtMemTypeDefault), A_dims_, A_data_, &ml_value);
+    feeds.insert(std::make_pair("A", ml_value));
+
+    CreateMLValue<T>(TestCPUExecutionProvider()->GetAllocator(0, OrtMemTypeDefault), B_dims_, B_data_, &ml_value);
+    feeds.insert(std::make_pair("B", ml_value));
+
+    CreateMLValue<float>(TestCPUExecutionProvider()->GetAllocator(0, OrtMemTypeDefault), {1}, B_scale_, &ml_value);
+    feeds.insert(std::make_pair("b_scale", ml_value));
+
+    CreateMLValue<T>(TestCPUExecutionProvider()->GetAllocator(0, OrtMemTypeDefault), {1}, B_zero_point_, &ml_value);
+    feeds.insert(std::make_pair("b_zero_point", ml_value));
+
```

Comment:
What's the reason why we can't use the same feeds from FillFeedsAndOutputNames used in Compute?

Reply:
it need to keep the .onnx file with exactly same input/output names as the schema definition's input/output. Happen to the case for this test.